infra-library.sh

```diff
@@ -117,23 +117,37 @@ function create_kind_test_cluster() {
 }
 
 # Create a GKE test cluster with kubetest2 and run the test command.
-# Parameters: $1 - custom flags defined in kntest
-#             $2 - test command to run after the cluster is created (optional)
+# Parameters: $1 - custom flags defined in kubetest2
+#             $2 - custom flags to pass directly to gcloud
+#             $3 - test command to run after the cluster is created (optional)
 function create_gke_test_cluster() {
   local -n _custom_flags=$1
-  local -n _test_command=$2
+  local -n _extra_gcloud_flags=$2
+  local -n _test_command=$3
 
   # We are disabling logs and metrics on Boskos Clusters by default as they are not used. Manually set ENABLE_GKE_TELEMETRY to true to enable telemetry
   # and ENABLE_PREEMPTIBLE_NODES to true to create preemptible/spot VMs. VM Preemption is a rare event and shouldn't be distruptive given the fault tolerant nature of our tests.
-  local extra_gcloud_flags=""
   if [[ "${ENABLE_GKE_TELEMETRY:-}" != "true" ]]; then
-    extra_gcloud_flags="${extra_gcloud_flags} --logging=NONE --monitoring=NONE"
+    _extra_gcloud_flags="${_extra_gcloud_flags} --logging=NONE --monitoring=NONE"
   fi
 
   if [[ "${ENABLE_PREEMPTIBLE_NODES:-}" == "true" ]]; then
-    extra_gcloud_flags="${extra_gcloud_flags} --preemptible"
+    _extra_gcloud_flags="${_extra_gcloud_flags} --preemptible"
+  fi
+  
+  if [[ "${USE_KUBETEST2_DIRECTLY:-}" == "true" ]]; then
+    _extra_gcloud_flags="${_extra_gcloud_flags[@]} --quiet"
+    kubetest2 gke "${_custom_flags[@]}" --up --down \
```

Comment:
I would like to avoid people needing to install an appropriate version of kubetest2 on their machines.

I'm okay to use kubetest2 directly on PROW, but on developer machines it should be called via:

```shell
go_run sigs.k8s.io/kubetest2@v0.0.0-20220903000027-a46ae8d7fa37 gke "${_custom_flags[@]}" --up --down \
```

Reply:
kubetest2 can't be run by developers as it needs access to GCP and it will try to talk to boskos endpoint which is isn't reachable from dev laptops.

Reply:
https://github.com/knative/test-infra/issues/2681

Reply:
Of course, it could. People could use different GCP project, right?